Assets/Tests/InputSystem/CoreTests_Editor_InputActionAssetManager.cs

```diff
@@ -0,0 +1,134 @@
+    {
+        InputActionAsset asset = null;
+        try
+        {
+            asset = ScriptableObject.CreateInstance<InputActionAsset>();
+            Assert.Throws<Exception>(() =>
+            {
+                using (var x = new InputActionAssetManager(asset))
+                {
+                }
+            });
+        }
+        finally
+        {
+            UnityEngine.Object.DestroyImmediate(asset);
+        }
+    }
+
+    [Test]
+    [Category("Editor")]
+    public void Editor_InputActionAssetManager_SerializedObjectIsAWorkingCopyOfImportedAsset()
+    {
+        var asset = AssetDatabaseUtils.CreateAsset<InputActionAsset>();
+        using (var sut = new InputActionAssetManager(asset))
+        {
+            Assert.That(sut.serializedObject.targetObject == null, Is.False);
+            Assert.That(sut.serializedObject.targetObject == asset, Is.False);
+        }
+    }
+
+    [Test]
+    [Category("Editor")]
+    public void Editor_InputActionAssetManager_isDirtyReflectsWhetherWorkingCopyIsDifferentFromImportedAsset()
+    {
+        // Start from an asset, since we haven't done modifications expect it to not be marked dirty
+        var asset = AssetDatabaseUtils.CreateAsset<InputActionAsset>();
+        using (var sut = new InputActionAssetManager(asset))
+        {
+            Assert.That(sut.dirty, Is.False);
+
+            // Add a map to editable object and apply changes, expect it to not be dirty since we haven't explicitly updated dirty state
+            var mapProperty = InputActionSerializationHelpers.AddActionMap(sut.serializedObject);
+            sut.ApplyChanges();
+            Assert.That(sut.dirty, Is.False); // TODO Note: dirty flag is not updated only by applying changes
+
+            // Update the dirty state, finally expect it to indicate that it has changed
+            sut.UpdateAssetDirtyState();
+            Assert.That(sut.dirty, Is.True);
+
+            // Remove the map we previously added and apply changes, expect it to still be dirty since we haven't explicitly updated dirty state
+            var editedAsset = sut.serializedObject.targetObject as InputActionAsset;
+            InputActionSerializationHelpers.DeleteActionMap(sut.serializedObject, editedAsset.actionMaps[0].id);
+            sut.ApplyChanges();
+            Assert.That(sut.dirty, Is.True);
+
+            // Update the dirty state, expect it to be false since even though we carried out changes we are now back on square one
+            sut.UpdateAssetDirtyState();
+            Assert.That(sut.dirty, Is.False);
+        }
+    }
+
+    [Test]
+    [Category("Editor")]
+    public void Editor_InputActionAssetManager_CanMoveAssetOnDisk()
+    {
+        const string filename = "InputAsset." + InputActionAsset.Extension;
+        var directoryBeforeMove = AssetDatabaseUtils.CreateDirectory();
+        var directoryAfterMove = AssetDatabaseUtils.RandomDirectoryPath();
```

Comment:
Switching to random paths was not strictly necessary but the reason why I introduced that utility is the asset file mess that currently exists in our tests leaving all sorts of files as left overs. Hence why I created AssetDatabaseUtil.cs. Random paths may be annoying but originated as that to avoid all the clashes test cases caused when not removing things properly in teardown. The added util takes care of this and guarantees that there are never clashes even if file cleanup would fail.